File: app/src/main/java/com/example/runtimetracker/TaskRun.java
package com.example.runtimetracker;

import java.sql.Date;

public class TaskRun extends Run {
    public AddEditRunActivity.TASK_TYPE taskType;

    public TaskRun(long id, String runTime, float runDistance, Date runDate, AddEditRunActivity.TASK_TYPE taskType) {
        super(id, runTime, runDistance, runDate);
        this.taskType = taskType;
    }

    public TaskRun(AddEditRunActivity.TASK_TYPE taskType) {
        this.taskType = taskType;
    }
}

File: app/src/main/java/com/example/runtimetracker/TaskRun.kt
package com.example.runtimetracker

import com.example.runtimetracker.AddEditRunActivity.TASK_TYPE
import java.sql.Date

class TaskRun : Run {
    var taskType: TASK_TYPE

    constructor(id: Long, runTime: String, runDistance: Float, runDate: Date?, taskType: TASK_TYPE) : super(id, runTime, runDistance, runDate) {
        this.taskType = taskType
    }

    constructor(taskType: TASK_TYPE) {
        this.taskType = taskType
    }
}